### PR TITLE
fix: rss feed for changelog

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
## Description

Fix Rss-Feed for the changelog
## Related Issue(s)

Fixes #

## How to test

call the route `/changelog/rss.xml`

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/1248"></a>

### CODE REVIEW

File: src/routes/changelog/rss.xml.ts [hunk 1]
```diff
@@ -33,10 +36,26 @@ export const get: import("@sveltejs/kit").RequestHandler = ({ context }) => {
...
     headers: {
       "Cache-Control": `max-age=0, s-max-age=${ttlInMin * 60}`,
       "Content-Type": "application/rss+xml",
     },
   };
 };
+
+const getChangelogs = async (): Promise<ChangelogEntry[]> => {
+  const changelogEntries = await Promise.all(
+    Object.entries(import.meta.glob("/src/contents/changelog/*.md")).map(
```
Comment (Cahllagerfeld) [inline src/routes/changelog/rss.xml.ts:49]: In order to use that `import.meta` I had to adjust the `tsconfig`

File: tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
 	"compilerOptions": {
+		"module": "ES2020",
```
Comment (Cahllagerfeld) [inline tsconfig.json:3]: See comment on `import.meta`

I'm not qutie sure if this has any side-effects on the `target: es2019` though

File: src/routes/changelog/rss.xml.ts [hunk 3]
```diff
@@ -33,10 +36,26 @@ export const get: import("@sveltejs/kit").RequestHandler = ({ context }) => {
...
-    body: rssDocument,
+    body: rssDocument(logs),
     headers: {
       "Cache-Control": `max-age=0, s-max-age=${ttlInMin * 60}`,
       "Content-Type": "application/rss+xml",
     },
   };
 };
+
+const getChangelogs = async (): Promise<ChangelogEntry[]> => {
```
Comment (Cahllagerfeld) [inline src/routes/changelog/rss.xml.ts:47]: This method is a slightly adjusted version of the changelog-function from `hooks.ts`

**Review by Cahllagerfeld — COMMENTED** [review]
Just a few comments on what I did and my thoughts

**Comment by Cahllagerfeld** [comment]
I couldnt find a related issue for that, should I open one for consistancy, or is it just a "on-the-fly"-quick-fix :D

File: src/routes/changelog/rss.xml.ts [hunk 4]
```diff
@@ -1,4 +1,8 @@
-export const get: import("@sveltejs/kit").RequestHandler = ({ context }) => {
+import type { ChangelogEntry } from "../../types/changelog-entry.type";
+
+export const get: import("@sveltejs/kit").RequestHandler = async ({
+  locals,
```
Comment (mikenikles) [inline src/routes/changelog/rss.xml.ts:4]: @Cahllagerfeld The `context` was available in the older version of SvelteKit. The concept changed slightly in newer versions and we can now use `locals`. See `hooks.ts` where I refactored it as part of the SvelteKit upgrade.

**Review by mikenikles — APPROVED** [review]
Approving this so you can merge when you get up and if you agree with my changes.

File: src/routes/changelog/rss.xml.ts [hunk 5]
```diff
@@ -29,7 +32,8 @@ export const get: import("@sveltejs/kit").RequestHandler = ({ context }) => {
         'src="https://www.gitpod.io/'
       )}']]></description>
     </item>`
-   )}
+     )
+     .join("")}
```
Comment (mikenikles) [inline src/routes/changelog/rss.xml.ts:36]: This removes the `,` we'd get without it between `</item>,<item>`. That was a bug in the old code too but is now fixed.

**Comment by Cahllagerfeld** [comment]
LGTM! 👍

**Pull request merged**